### PR TITLE
Support for Immature Coin Base Transaction

### PR DESCRIPTION
**Pull request by sageprogrammer**
Closes: #321

### CODE REVIEW

**Comment by sageprogrammer** [comment]
closes #321

**Comment by NicolasDorier** [comment]
It seems like a good solution. I a wondering if `Total` should take into account immature transactions.

**Comment by NicolasDorier** [comment]
Can you update the doc in `docs/API.md`?

**Comment by NicolasDorier** [comment]
Looking at the code, the Total already should not report immature coins. So something else might be wrong.

File: docker-compose.regtest.yml [hunk 1]
```diff
@@ -54,10 +54,6 @@ services:
   rabbitmq:
     image: rabbitmq:3-management
     hostname: rabbitmq
-    environment:
```
Comment (NicolasDorier) [inline docker-compose.regtest.yml:57]: why rmoving this?

**Comment by NicolasDorier** [comment]
Actually, I manage to reproduce the immature issue. Trying to understand why my code did not work, I thought taking care of it.

**Comment by NicolasDorier** [comment]
I will slightly change the API, but fix is good thanks a lot!

File: docker-compose.regtest.yml (hunk 1 above)
Comment (sageprogrammer) [reply docker-compose.regtest.yml:57]: Actually when I tried to run rabbitmq via docker-compose, it was complaining that 
rabbitmq_1          | error: RABBITMQ_DEFAULT_PASS is set but deprecated
rabbitmq_1          | error: RABBITMQ_DEFAULT_USER is set but deprecated
rabbitmq_1          | error: RABBITMQ_DEFAULT_VHOST is set but deprecated
rabbitmq_1          | error: deprecated environment variables detected
rabbitmq_1          | 
rabbitmq_1          | Please use a configuration file instead; visit https://www.rabbitmq.com/configure.html to learn more


Also I found that even if we do not provide any default ones, according to https://www.rabbitmq.com/access-control.html#default-state, It still uses same values.

**Comment by NicolasDorier** [comment]
tests and doc, and slight API change on https://github.com/dgarage/NBXplorer/pull/324 let me know.

**Pull request merged**